test/index.js

```diff
@@ -1091,6 +1065,114 @@ describe('PubSub', function() {
     });
   });
 
+  describe('getClient_', function() {
+    var CONFIG = {
+      client: 'SubscriberClient',
+      reqOpts: {},
+      gaxOpts: {},
+    };
+
+    beforeEach(function() {
+      pubsub.config = CONFIG;
+    });
+
+    it('should cache the client', function(done) {
+      pubsub.getClient_(CONFIG, function(err, client1) {
+        assert.strictEqual(null, err);
+        pubsub.getClient_(CONFIG, function(err, client2) {
+          assert.strictEqual(null, err);
+          assert.strictEqual(client1, client2);
+          done();
+        });
+      });
+    });
+  });
+
+  describe('request', function() {
+    var CONFIG = {
+      client: 'SubscriberClient',
+      method: 'fakeMethod',
+      reqOpts: {a: 'a'},
+      gaxOpts: {},
+    };
+
+    beforeEach(function() {
+      delete pubsub.projectId;
+
+      pubsub.auth = {
+        getProjectId: function(callback) {
+          callback(null, PROJECT_ID);
+        },
+      };
+
+      fakeUtil.replaceProjectIdToken = function(reqOpts) {
+        return reqOpts;
+      };
+
+      pubsub.config = CONFIG;
+    });
+
+    it('should get the project id', function(done) {
+      pubsub.auth.getProjectId = function(callback) {
+        assert.strictEqual(typeof callback, 'function');
+        done();
+      };
+
+      pubsub.request(CONFIG, assert.ifError);
+    });
+
+    it('should return auth errors to the callback', function(done) {
+      var error = new Error('err');
+
+      pubsub.auth.getProjectId = function(callback) {
+        callback(error);
+      };
+
+      pubsub.request(CONFIG, function(err) {
+        assert.strictEqual(err, error);
+        done();
+      });
+    });
+
+    it('should replace the project id token on reqOpts', function(done) {
+      fakeUtil.replaceProjectIdToken = function(reqOpts, projectId) {
+        assert.deepEqual(reqOpts, CONFIG.reqOpts);
+        assert.strictEqual(projectId, PROJECT_ID);
+        done();
+      };
+
+      pubsub.request(CONFIG, assert.ifError);
+    });
+
+    it('should do nothing if sandbox env var is set', function(done) {
+      global.GCLOUD_SANDBOX_ENV = true;
+      pubsub.request(CONFIG, done); // should not fire done
+      global.GCLOUD_SANDBOX_ENV = false;
+      done();
+    });
+  });
+
+  describe('request', function() {
```

Comment:
For `request`, I think we need new tests, and we can re-use one from above. Here's what I would do:

- should call getClient_ correctly (checks that `config` object is received by `getClient_`)
- should return an error from getClient_
- **re-used test**: should replace the project id token on reqOpts
- should call client method correctly (see that it gets the correct `reqOpts`, `config.gaxOpts`, and `callback`)